common/app/services/ParameterStore.scala

```diff
@@ -0,0 +1,63 @@
+package services
+
+import com.amazonaws.services.simplesystemsmanagement.model.{GetParameterRequest, GetParametersByPathRequest}
+import com.amazonaws.services.simplesystemsmanagement.{AWSSimpleSystemsManagement, AWSSimpleSystemsManagementClientBuilder}
+import conf.Configuration
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+
+class ParameterStore(region: String) {
+
+  private lazy val client: AWSSimpleSystemsManagement = Configuration.aws.credentials.map { credentials =>
+    AWSSimpleSystemsManagementClientBuilder
+      .standard()
+      .withCredentials(credentials)
+      .withRegion(region)
+      .build()
+  }.getOrElse(throw new RuntimeException("Failed to initialize AWSSimpleSystemsManagement"))
+
+  def get(key: String): String = {
+    val parameterRequest = new GetParameterRequest().withWithDecryption(true).withName(key)
+    client.getParameter(parameterRequest).getParameter.getValue
+  }
+
+  def getPath(path: String): Map[String, String] = {
+
+    def unwrapQuotedString(input: String) = {
+      val quotedString = "\"(.*)\"".r
+      input match {
+        case quotedString(content) => content
+        case content => content
+      }
+    }
+
+    @tailrec
+    def pagination(accum: Map[String, String], nextToken: Option[String]): Map[String, String] = {
+
+      // Only possible to get a maximum of 10 results when this was written
+      val maxResults = 10
+
+      val parameterRequest = new GetParametersByPathRequest()
+        .withWithDecryption(true)
+        .withPath(path)
+        .withRecursive(false)
+        .withMaxResults(maxResults)
```

Comment:
Is this parameter required? If not would it be better not to specify it so if amazon decides to change how many results is being sent we would not need to change our code?

Reply:
I wanted to ensure the maximum was used. I think it defaults to 10 anyway so can remove.